src/python/WMArchive/Service/Data.py

```diff
@@ -29,6 +29,14 @@
 from WMArchive.Service.Manager import WMArchiveManager
 from WMArchive.Utils.Regexp import PAT_UID, PAT_QUERY, PAT_INFO
 
+def results(result):
+    "Return results as a list data type. Set proper status in case of failures"
+    if 'status' in result and 'Not supported' in result['status']:
+        cherrypy.response.status = "406 Not Acceptable"
```

Comment:
Although, this is fine but could you just set it as integer status code? That is what generally expected all other status in WMCore code.